### ISSUE TITLE
Make use of SFTP API for `incus file` operations

### ISSUE THREAD

**Issue by stgraber** [lxc/incus#1176]
Currently all the `incus file` operations are done through the individual REST API endpoints, which Incus then internally converts to SFTP interactions. This isn't exactly the most flexible or efficient.

It'd be good to change all the `incus file` operations to instead use `GetInstanceFileSFTP` to fetch a Go SFTP client and then perform the relevant filesystem operations through it.

- [x] create
- [ ] delete
- [x] push
- [x] pull
- [ ] edit

**Comment by HassanAlsamahi** [comment]
Can I work on this? If that is okay, should the change be done in client/incus_instances and replace the REST API part of the functions CreateInstanceFile..etc with the SFTP part? or directly in the cmd/incus/files? I believe the first but wanted to verify.

**Comment by stgraber** [comment]
Assigned it to you!

The change should be done directly in `cmd/incus/file.go`, the Go client package should be a mostly clear mapping to the API so we don't want it to get too smart on that front.

**Comment by stgraber** [comment]
Hey @HassanAlsamahi, how are things going with this one?

**Comment by HassanAlsamahi** [comment]
Hey @stgraber . I made some progress but I didn't push a commit yet as I got busy and some personal stuff. I'll push a commit very soon for one of the operations, followed by the other operations. Thanks for asking and sorry for the delay.

**Comment by stgraber** [comment]
No worries, thanks for the update!

**Comment by stgraber** [comment]
@HassanAlsamahi any luck with the other ones?

**Comment by HassanAlsamahi** [comment]
Hey @stgraber . Sorry I was on vacation in October totally logged off. Please check the latest PR and let me know if anything is needed regarding it.

**Comment by stgraber** [comment]
Thanks!

**Comment by HassanAlsamahi** [comment]
Hey @stgraber , I have a question. Does the functions `recursivePullFile` and `recursivePushFile` need also to be modified to use the SFTP package instead of `buf, resp, err := d.GetInstanceFile(inst, p)` and `err = d.CreateInstanceFile(inst, targetPath, args)`  respectively?

**Comment by stgraber** [comment]
Hey @HassanAlsamahi, how's it going with this one?

**Comment by HassanAlsamahi** [comment]
Hey @stgraber . It's getting so busy in here, I had the code updated for the pull operation but didn't get the chance to organize, validate, and commit until now. Please check the latest PR and let me know if any modifications needed.

Pending now is
- [ ] edit
- [ ] recursivePullFile
- [ ] recursivePushFile

**Comment by HassanAlsamahi** [comment]
Hey @stgraber . I noticed that the code of the edit command is based on FilePush and FilePull functions, which are already changed to the SFTP package. Is there anything that needs to be done for the Edit command? Did I miss anything?

**Comment by stgraber** [comment]
No, you're right, I don't expect we need any change to that command

**Comment by stgraber** [comment]
So we still have:
 - `GetInstanceFile` in `recursivePullFile` and `recursiveMkdir`
 - `CreateInstanceFile` in `recursivePushFile` and `recursiveMkdir`
